### PR TITLE
Fit_generator: Create validation data queue only once

### LINKED ISSUE #8842: Queue of validation generator in fit_generator
Hi,

I am using `fit_generator` with a data_generator for validation_data.

The code is:

`history=model.fit_generator(generator=generator_trai,steps_per_epoch=200,epochs=2,validation_data=generator_val,validation_steps=15)`

I set `validation_steps=15`. Because of the default queue_size of 10 I expect the generator to add 25 batches to the queue in the first epoch. 15 of 25 for doing the validation in the first epoch and further 10 for filling up the queue. So far so good. But I am wondering why the generator adds futher 25 batches in the second epoch. In my opinion it should only add the 15 batches that are used for validation in second epoch.

Why is this happening? 

The Training Generator works as supposed. Adding 210 batches in first epoch and futher 200 batches in second epoch.

Thanks in advance.

### PR DESCRIPTION
**Pull request by jgrnt**
As reported in #8842 the enqueuer is recreated during every evaluation of the validation set.
This creates the enqueuer in `fit_generator` and invokes  `evaluate_generator` with the resulting generator while disabling queuing.

Fixes #8842

### CODE REVIEW

File: keras/engine/training.py [hunk 1]
```diff
@@ -2100,27 +2100,45 @@ def generate_arrays_from_file(path):
...
-            for cbk in callbacks:
-                cbk.validation_data = val_data
         enqueuer = None
+        val_enqueuer = None
 
         try:
+            if do_validation:
+                if val_gen:
+                    if workers > 0:
+                        if is_sequence:
```
Comment (Dref360) [inline keras/engine/training.py]: need to recompute this, since the validation_data may not be a Sequence

File: keras/engine/training.py [hunk 2]
```diff
@@ -2219,8 +2235,12 @@ def generate_arrays_from_file(path):
                     break
 
         finally:
-            if enqueuer is not None:
-                enqueuer.stop()
+            try:
```
Comment (Dref360) [inline keras/engine/training.py]: Why two nested try catch ?

Comment (jgrnt) [reply keras/engine/training.py]: If an exception is raised in ```enqueuer.stop```, we would not stop the ```val_enqueuer```.
I guess alternatively one could make the enqueuers context managers.

File: keras/engine/training.py (hunk 1 above)
Comment (jgrnt) [reply keras/engine/training.py]: Thanks for spotting the error.
Fixed

**Comment by fchollet** [comment]
Thanks for the PR and the reviews. @Dref360, do you LGTM?

**Comment by Dref360** [comment]
LGTM
